src/transformers/modeling_utils.py

```diff
@@ -885,10 +923,15 @@ def generate(
 
         # create attention mask if necessary
         # TODO (PVP): this should later be handled by the forward fn() in each model in the future see PR 3140
-        if (attention_mask is None) and (pad_token_id is not None) and (pad_token_id in input_ids):
-            attention_mask = input_ids.ne(pad_token_id).long()
-        elif attention_mask is None:
-            attention_mask = input_ids.new_ones(input_ids.shape)
+        if (decoder_attention_mask is None) and (pad_token_id is not None) and (pad_token_id in decoder_input_ids):
+            decoder_attention_mask = decoder_input_ids.ne(pad_token_id).long()
+        elif decoder_attention_mask is None:
+            decoder_attention_mask = decoder_input_ids.new_ones(decoder_input_ids.shape)
+        if self.config.is_encoder_decoder:
+            if (encoder_attention_mask is None) and (pad_token_id is not None) and (pad_token_id in encoder_input_ids):
+                encoder_attention_mask = encoder_input_ids.ne(pad_token_id).long()
+            elif encoder_attention_mask is None:
+                encoder_attention_mask = encoder_input_ids.new_ones(encoder_input_ids.shape)
```

Comment:
If you're going to make the mask anyway, the `(pad_token_id in encoder_input_ids)` check is unnecessary.